### PR TITLE
Protocol type version fix

### PR DESCRIPTION
**Pull request by janstaelensskyline**
* Allows tagging with pre-release versions. (Note, will still fail until further fixes in upload API)
* Improves logging and exception logging for deployments and uploads
*

Note: Before release, first release Reusable Modules and adjust the NuGet Package for dmapp creation in the deploy action.

### CODE REVIEW

File: GitHubAction/UploadArtifactApi/HttpArtifactUploadApi.cs [hunk 1]
```diff
@@ -15,24 +17,28 @@ public HttpArtifactUploadApi(HttpClient httpClient)
...
+        CancellationToken cancellationToken, IPackagePresenter presenter)
     {
         string dmappFilePath = Path.Combine(Environment.GetEnvironmentVariable("GITHUB_WORKSPACE"), name);
-		File.WriteAllBytes(dmappFilePath, package);
-		FileStream fileStream = new FileStream(dmappFilePath, FileMode.Open, FileAccess.Read, FileShare.None);
+        File.WriteAllBytes(dmappFilePath, package);
+        FileStream fileStream = new FileStream(dmappFilePath, FileMode.Open, FileAccess.Read, FileShare.None);
 
-		using var formData = new MultipartFormDataContent();
+        StringBuilder sb = new StringBuilder();
```
Comment (MichielOda) [inline GitHubAction/UploadArtifactApi/HttpArtifactUploadApi.cs:31]:
```suggestion
```

Comment (MichielOda) [reply GitHubAction/UploadArtifactApi/HttpArtifactUploadApi.cs:31]: Unused variable

File: GitHubAction/UploadArtifactApi/HttpArtifactUploadApi.cs [hunk 2]
```diff
@@ -43,10 +49,10 @@ public HttpArtifactUploadApi(HttpClient httpClient)
 
         if (response.StatusCode is HttpStatusCode.Forbidden or HttpStatusCode.Unauthorized)
         {
-            throw new KeyException($"The upload api returned a {response.StatusCode} response.");
+            throw new KeyException($"The upload api returned a {response.StatusCode} response. Body:" + response?.Content?.ToString() ?? "null");
```
Comment (MichielOda) [inline GitHubAction/UploadArtifactApi/HttpArtifactUploadApi.cs:52]:
```suggestion
            throw new KeyException($"The upload api returned a {response.StatusCode} response. Body: {response.Content}");
```

File: GitHubAction/UploadArtifactApi/HttpArtifactUploadApi.cs [hunk 3]
```diff
@@ -43,10 +49,10 @@ public HttpArtifactUploadApi(HttpClient httpClient)
...
         }
 
-        throw new UploadPackageException($"The upload api returned a {response.StatusCode} response.");
+        throw new UploadPackageException($"The upload api returned a {response.StatusCode} response. Body:" + response?.Content?.ToString() ?? "null");
```
Comment (MichielOda) [inline GitHubAction/UploadArtifactApi/HttpArtifactUploadApi.cs:55]:
```suggestion
        throw new UploadPackageException($"The upload api returned a {response.StatusCode} response. Body: {response.Content}");
```

File: GitHubAction/UploadArtifactApi/HttpArtifactUploadApi.cs (hunk 2 above)
Comment (MichielOda) [reply GitHubAction/UploadArtifactApi/HttpArtifactUploadApi.cs:52]: Content will never be null
image

File: GitHubAction/Package.Application/PackageService.cs [hunk 4]
```diff
@@ -19,9 +19,9 @@ public async Task<CreatedPackage> CreatePackageAsync(LocalPackageConfig localPac
         return await _packageBuilder.CreatePackageAsync(localPackageConfig);
     }
 
-    public async Task<UploadedPackage> UploadPackageAsync(CreatedPackage createdPackage, string key)
+    public async Task<UploadedPackage> UploadPackageAsync(CreatedPackage createdPackage,string catalogVersion, string key)
```
Comment (MichielOda) [inline GitHubAction/Package.Application/PackageService.cs:22]:
```suggestion
    public async Task<UploadedPackage> UploadPackageAsync(CreatedPackage createdPackage, string catalogVersion, string key)
```

**Review by MichielOda — CHANGES_REQUESTED** [review]

File: GitHubAction/GitHubAction.Console/Extensions/ApiExtensions.cs [hunk 5]
```diff
@@ -31,6 +31,7 @@ public static void AddApis(this IServiceCollection services)
         {
             var httpClient = s.GetRequiredService<IHttpClientFactory>().CreateClient(nameof(HttpArtifactUploadApi));
             var options = s.GetService<ApiOptions>();
+            var presenter = s.GetService<ConsolePackagePresenter>();
```
[the remark this answers is not in the document]
Comment (janstaelensskyline) [reply GitHubAction/GitHubAction.Console/Extensions/ApiExtensions.cs:34]: removed.

**Review by MichielOda — DISMISSED** [review]

**Pull request merged**